include/AssetManager.h

```diff
@@ -27,8 +27,8 @@ class AssetManager
     void createProjectile(Vector2D pos, Vector2D velocity, int scale, int range, int speed, const char* texturePath, Entity* owner);
     void createPowerup(Vector2D pos, std::function<void (Entity*)> pickupFunc, std::string texturePath);
 
-    Vector2D calculateSpawnPosition();
-    PowerupType calculateType();
+    Vector2D calculateSpawnPosition(int width, int height);
```

Comment:
width and height as names are not immediately clear as meaning width/height of the object to be spawned, maybe replace with a Vector2D `size`.

Additionally, parameterizing the spawn area could be useful (currently the screen space).

Please also add documentation for this functions (brief, parameters, return value)